src/components/XDataGrid/CommonDataGrid.tsx

```diff
@@ -0,0 +1,88 @@
+import { DataGrid, DataGridProps, GridToolbar } from '@mui/x-data-grid';
+import { LinearProgress } from '@mui/material';
+import CustomNoRowsOverlay from './CustomNoRowsOverlay';
+import { GridValidRowModel } from '@mui/x-data-grid/models/gridRows';
+import { ReactElement, useEffect, useMemo, useState } from 'react';
+import { deepmerge } from '@mui/utils';
+import { useSnackMessage } from '@gridsuite/commons-ui';
+
+export default function CommonDataGrid<R extends GridValidRowModel>(
+    props: Omit<DataGridProps<R>, 'rows' | 'loading'> & {
+        //rows: Partial<DataGridProps<R>['rows']>;
+        loader: () => Promise<R[]>;
+    }
+): ReactElement {
```

Comment:
We didn't say we should use `FunctionalComponent` ?

Reply:
yes , but dunno how to use it with `function` keyword...

Reply:
We can't that's why we use arrow function